PaginatedDataTable loading indicator compliance with Material Specification

wer-mathurin opened the pull request:
This PR change the loading indicator of the paginated data table with a linear progress indicator like describe in the ticket (https://github.com/flutter/flutter/issues/102861)

List which issues are fixed by this PR. You must list at least one issue.
https://github.com/flutter/flutter/issues/102861
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

Commit:
FIX - #102861

Commit:
FIX - #102861

Commit:
Remove trailling space

Commit:
Remove trailling space

Commit:
FIX - Specify Type Annotation

wer-mathurin commented:
Hi @goderbauer,

just need some attention to continue the process !  I think this is a good addition to comply with Material Specification

chunhtai commented:
I also found this sentence in material design.
> A progress indicator can be used with sorting or pagination. Two cursors are displayed for demonstration purposes.

with a picture display linear progress indicator

![image](https://user-images.githubusercontent.com/47866232/170558487-8c8716b6-b59d-4cb9-8fd1-79367b718206.png)

I think this pr matches material design better than the circular indicator

chunhtai commented on packages/flutter/lib/src/material/paginated_data_table.dart:
```diff
@@ -315,37 +315,26 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
-      if (!column.numeric) {
-        haveProgressIndicator = true;
-        return const DataCell(CircularProgressIndicator());
+  bool _haveProgressIndicator(int firstRowIndex, int rowsPerPage) {
```
This can probably merged into get merge into _getRows? maybe make _getRows a more generic function like _updateRowsInCurrentPage to avoid widget.source.getRow called  twice for every index

wer-mathurin replied:
It will not get call twice for every index, because as soon we find a null, we exit.

wer-mathurin replied:
@chunhtai I did a mistake when trying to merge/rebase on the latest master version :-(
I just close this PR and open another one https://github.com/flutter/flutter/pull/110177

wer-mathurin replied:
Applying your comments.  And try to be closer to the material specification (by showing the current page when going to the next one....I didn't figure it out how to display the progress indicator without changing the page...I will need help or guidance on this one.

chunhtai reviewed (commented):
One thing I noticed is that the row will display old data when progress indicator shown on the screen. This pr only displays empty row.

itsjustkevin committed:
replace engine and framework mirror args with github username arg (#109239)

engine-flutter-autoroll committed:
Roll Flutter Engine from 98c0699c50e7 to 7b6c20674d11 (1 revision) (#109341)

Hixie committed:
API documentation cleanup (#108500)

fluttergithubbot committed:
Roll pub packages (#109348)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7b6c20674d11 to d8169a4b1d4f (1 revision) (#109347)

Hixie committed:
Actually run customer_testing shard on Windows (#109336)

chunhtai committed:
Change ClipboardStatusNofifier parameter in buildToolbar to ValueLise… (#108917)

engine-flutter-autoroll committed:
Roll Flutter Engine from d8169a4b1d4f to a5abd389e4f4 (2 revisions) (#109352)

gspencergoog committed:
Revert "Roll pub packages (#109348)" (#109363)

Committing to green the build.

stuartmorgan-g committed:
Fix Android platform view creation flow (#109232)

engine-flutter-autoroll committed:
Roll Flutter Engine from a5abd389e4f4 to 697d6eeb3516 (1 revision) (#109354)

Hixie committed:
Minor fix compendium (#107874)

engine-flutter-autoroll committed:
Roll Flutter Engine from 697d6eeb3516 to 1b3b3edac1b0 (5 revisions) (#109366)

Commit:
Revert "Fix Android platform view creation flow" (#109389)

Merging on red to green the build.

Hixie committed:
Fix HTML in skeletons (ampersands must be escaped) (#109360)

engine-flutter-autoroll committed:
Roll Flutter Engine from 1b3b3edac1b0 to c9a31b7ea4fe (1 revision) (#109368)

engine-flutter-autoroll committed:
Roll Plugins from a124dcb33b62 to 6c9738ea56f9 (5 revisions) (#109390)

engine-flutter-autoroll committed:
Roll Flutter Engine from c9a31b7ea4fe to cf8d7c01d66d (10 revisions) (#109394)

Commit:
[framework] document backdropfilter antipattern (#109340)

engine-flutter-autoroll committed:
Roll Flutter Engine from cf8d7c01d66d to 492cf91f2854 (1 revision) (#109396)

Commit:
Revert "Change ClipboardStatusNofifier parameter in buildToolbar to V… (#109388)

QuncCccccc committed:
Set `IconButton.visualDensity` default to `VisualDensity.standard` (#109349)

* Set IconButton's visualDensity to standard

Co-authored-by: Qun Cheng <quncheng@google.com>

engine-flutter-autoroll committed:
Roll Flutter Engine from 492cf91f2854 to 7261384fc04e (1 revision) (#109402)

flar committed:
add page delays to the fullscreen versions of the textfield perf benchmarks (#107746)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7261384fc04e to 02cbee29fa0e (2 revisions) (#109407)

Jasguerrero committed:
Disable jank metrics for web (#109356)

QuncCccccc committed:
Added documentation for AppBar actions (#109351)

* Added documentation for AppBar actions

Co-authored-by: Qun Cheng <quncheng@google.com>

engine-flutter-autoroll committed:
Roll Flutter Engine from 02cbee29fa0e to 18a290082614 (4 revisions) (#109411)

xvrh committed:
Fix Transporter app name in log after upload (#107414)

verath committed:
Free library even if proc lookup fails (#108312)

fluttergithubbot committed:
Roll pub packages (#109408)

engine-flutter-autoroll committed:
Roll Flutter Engine from 18a290082614 to 7509b0741152 (1 revision) (#109415)

QuncCccccc committed:
Revert "Set `IconButton.visualDensity` default to `VisualDensity.standard` (#109349)" (#109421)

This reverts commit 3070baf17581b50b6b9327fd2b682a6742db04e0.

engine-flutter-autoroll committed:
Roll Flutter Engine from 7509b0741152 to 53bbdd647692 (1 revision) (#109418)

christopherfujino committed:
Revert "Roll pub packages (#109408)" (#109424)

This reverts commit da036da69f8b1f2a2f227d3bb46ab30a7344776f.

engine-flutter-autoroll committed:
Roll Flutter Engine from 53bbdd647692 to d42fa7a8044b (4 revisions) (#109422)

christopherfujino committed:
[flutter_tools] Pin path_provider_android (#109429)

stuartmorgan-g committed:
Reland: Fix Android platform view creation flow (#109405)

engine-flutter-autoroll committed:
Roll Flutter Engine from d42fa7a8044b to 3747cf318c58 (7 revisions) (#109430)

engine-flutter-autoroll committed:
Roll Flutter Engine from 3747cf318c58 to bcecbae85ba1 (3 revisions) (#109434)

engine-flutter-autoroll committed:
Roll Flutter Engine from bcecbae85ba1 to adbbadf17708 (1 revision) (#109436)

engine-flutter-autoroll committed:
Roll Flutter Engine from adbbadf17708 to 643742e4fb97 (1 revision) (#109438)

engine-flutter-autoroll committed:
Roll Flutter Engine from 643742e4fb97 to 913f6b23e229 (1 revision) (#109442)

engine-flutter-autoroll committed:
Roll Flutter Engine from 913f6b23e229 to d35771a75191 (2 revisions) (#109444)

engine-flutter-autoroll committed:
Roll Flutter Engine from d35771a75191 to e20ba0df407c (1 revision) (#109445)

engine-flutter-autoroll committed:
Roll Flutter Engine from e20ba0df407c to 34933f96b7ea (1 revision) (#109447)

engine-flutter-autoroll committed:
Roll Flutter Engine from 34933f96b7ea to 4f921bbd0093 (1 revision) (#109452)

engine-flutter-autoroll committed:
Roll Flutter Engine from 4f921bbd0093 to f29bc623e68d (1 revision) (#109454)

engine-flutter-autoroll committed:
Roll Plugins from 6c9738ea56f9 to 9fb7654f3aa2 (4 revisions) (#109456)

engine-flutter-autoroll committed:
Roll Flutter Engine from f29bc623e68d to de118a55ade0 (1 revision) (#109457)

yusuf-goog committed:
Updating expiring cirrus key. (#109461)

engine-flutter-autoroll committed:
Roll Flutter Engine from de118a55ade0 to c354e0e04cea (2 revisions) (#109462)

engine-flutter-autoroll committed:
Roll Flutter Engine from c354e0e04cea to d1b18b5d6811 (1 revision) (#109466)

goderbauer committed:
Keep `dirty` manipulations private to `Element` base class (#109401)

jmagman committed:
Remove deprecated Ruby File.exists? in helper script (#109428)

gspencergoog committed:
Revert "Keep `dirty` manipulations private to `Element` base class (#109401)" (#109481)

Landing on red to green the build.

engine-flutter-autoroll committed:
Roll Flutter Engine from d1b18b5d6811 to a2f3bd58ce73 (1 revision) (#109472)

Renzo-Olivares committed:
Single tap on the previous selection should toggle the toolbar on iOS… (#108913)

engine-flutter-autoroll committed:
Roll Flutter Engine from a2f3bd58ce73 to fb19700742c3 (6 revisions) (#109482)

engine-flutter-autoroll committed:
Roll Flutter Engine from fb19700742c3 to 2180e62f6c7e (1 revision) (#109483)

engine-flutter-autoroll committed:
Roll Flutter Engine from 2180e62f6c7e to cd1a1cb757f5 (1 revision) (#109488)

engine-flutter-autoroll committed:
Roll Flutter Engine from cd1a1cb757f5 to 5cc5c5f04f0a (1 revision) (#109490)

engine-flutter-autoroll committed:
Roll Flutter Engine from 5cc5c5f04f0a to 18dca3de7496 (2 revisions) (#109493)

engine-flutter-autoroll committed:
Roll Flutter Engine from 18dca3de7496 to 83771593cb55 (1 revision) (#109505)

Hixie committed:
Cleaner test.dart output. (#109206)

engine-flutter-autoroll committed:
Roll Flutter Engine from 83771593cb55 to f49a617535af (1 revision) (#109509)

godofredoc committed:
Update README.md (#109506)

limonadev committed:
Add onOpened callback to PopupMenuButton (#103753)

engine-flutter-autoroll committed:
Roll Flutter Engine from f49a617535af to cc9f4c63b7e5 (1 revision) (#109518)

engine-flutter-autoroll committed:
Roll Flutter Engine from cc9f4c63b7e5 to 4656c2f46cad (1 revision) (#109524)

engine-flutter-autoroll committed:
Roll Flutter Engine from 4656c2f46cad to 16fb19e72f88 (1 revision) (#109526)

engine-flutter-autoroll committed:
Roll Flutter Engine from 16fb19e72f88 to 9c3c233e2639 (1 revision) (#109532)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9c3c233e2639 to f0c3829e90ce (1 revision) (#109536)

engine-flutter-autoroll committed:
Roll Plugins from 9fb7654f3aa2 to 094f1c3ad94a (7 revisions) (#109543)

engine-flutter-autoroll committed:
Roll Flutter Engine from f0c3829e90ce to 8c019cdd446f (2 revisions) (#109548)

engine-flutter-autoroll committed:
Roll Flutter Engine from 8c019cdd446f to 198b0051a5a2 (1 revision) (#109552)

engine-flutter-autoroll committed:
Roll Flutter Engine from 198b0051a5a2 to 33fccf564973 (1 revision) (#109557)

goderbauer committed:
Reland "Keep `dirty` manipulations private to `Element` base class (#109401)" (#109562)

engine-flutter-autoroll committed:
Roll Flutter Engine from 33fccf564973 to b4cbebe3887e (3 revisions) (#109564)

bernaferrari committed:
Refactor StrokeAlign to allow double values. (#108339)

yaakovschectman committed:
Add Compilation DeviceLab test for Windows (#109540)

* Add Windows compilation test (hello_world_windows__compile)

* Add devicelab target to .ci.yaml

* Bringup on new target

* Add license to generated files

* Indicate test ownership

* Delete asset from repo

* Add Windows compilation test (hello_world_windows__compile)

* Add devicelab target to .ci.yaml

* Bringup on new target

* Add license to generated files

* Indicate test ownership

* Delete icon

* Use path.basename

Re triggering tests

* Update test owner to desktop team

engine-flutter-autoroll committed:
Roll Flutter Engine from b4cbebe3887e to bf9f07b7b573 (1 revision) (#109566)

engine-flutter-autoroll committed:
Roll Flutter Engine from bf9f07b7b573 to 2c3efa0beb24 (1 revision) (#109570)

engine-flutter-autoroll committed:
Roll Flutter Engine from 2c3efa0beb24 to 90477c21aab5 (2 revisions) (#109575)

GaryQian committed:
Update `flutter.gradle` AGP to 7.2.0 and bump default NDK version (#109211)

engine-flutter-autoroll committed:
Roll Flutter Engine from 90477c21aab5 to de2640e525d9 (2 revisions) (#109582)

engine-flutter-autoroll committed:
Roll Flutter Engine from de2640e525d9 to 90d845191153 (2 revisions) (#109584)

xu-baolin committed:
Fix Text selection handler disappear when dragged to new words (#108808)

engine-flutter-autoroll committed:
Roll Flutter Engine from 90d845191153 to bbe0038e2a36 (1 revision) (#109586)

engine-flutter-autoroll committed:
Roll Flutter Engine from bbe0038e2a36 to 280e9a7becad (1 revision) (#109587)

Commit:
Revert "Refactor StrokeAlign to allow double values." (#109591)

engine-flutter-autoroll committed:
Roll Flutter Engine from 280e9a7becad to 8a1ab4692ae0 (1 revision) (#109597)

TahaTesser committed:
Fix `CupertinoListTile``s onTap with delay throws exception (#109038)

maheshj01 committed:
Pass fit property to RenderIndexedStack (#109295)

engine-flutter-autoroll committed:
Roll Flutter Engine from 8a1ab4692ae0 to db741c85ad09 (1 revision) (#109611)

engine-flutter-autoroll committed:
Roll Plugins from 094f1c3ad94a to d18079b41616 (9 revisions) (#109615)

fzyzcjy committed:
Fix that RenderEditable (TextField) ignores offset in painting, making text selections shifted when offset is nonzero (#109287)

Small fix in text layout math, unlikely to affect most normal usages.

camsim99 committed:
Add Spellcheck to EditableText (Android) (#109334)

Commit:
Revert "Add Spellcheck to EditableText (Android) (#109334)" (#109631)

This reverts commit ff9fe35682e10d1067d918770048d28a1f462e0f.

engine-flutter-autoroll committed:
Roll Flutter Engine from db741c85ad09 to 688624b057db (1 revision) (#109617)

olof-dev committed:
Change "Unicode scalar values" to "Unicode grapheme clusters" in maxLength docs (#106275)

engine-flutter-autoroll committed:
Roll Flutter Engine from 688624b057db to 69ee0c7094f3 (12 revisions) (#109634)

gspencergoog committed:
Some changes needed to PlatformMenuBar before the MenuBar implementation change lands. (#109556)

Hixie commented:
@wer-mathurin Thanks for your contribution! Do you have the time and interest to address @chunhtai's review comments above? It's perfectly fine if you don't, we can mark the issue as having an incomplete PR and eventually someone else will take it over.

engine-flutter-autoroll committed:
Roll Flutter Engine from 69ee0c7094f3 to eb7428a51ddd (3 revisions) (#109639)

engine-flutter-autoroll committed:
Roll Flutter Engine from eb7428a51ddd to 028a5bcebfab (3 revisions) (#109647)

xu-baolin committed:
Scrollbar respect the NeverScrollableScrollPhysics physics (#109609)

LongCatIsLooong committed:
Fix a type casting error in text_input.dart (#109635)

engine-flutter-autoroll committed:
Roll Flutter Engine from 028a5bcebfab to 246f24589107 (1 revision) (#109651)

engine-flutter-autoroll committed:
Roll Flutter Engine from 246f24589107 to cd35ae3706a4 (3 revisions) (#109653)

engine-flutter-autoroll committed:
Roll Flutter Engine from cd35ae3706a4 to 2e6d4dc5b513 (1 revision) (#109657)

engine-flutter-autoroll committed:
Roll Flutter Engine from 2e6d4dc5b513 to 4024ff17ac5f (3 revisions) (#109660)

engine-flutter-autoroll committed:
Roll Flutter Engine from 4024ff17ac5f to fa91de4b2072 (1 revision) (#109662)

engine-flutter-autoroll committed:
Roll Flutter Engine from fa91de4b2072 to e480e68d34a7 (1 revision) (#109665)

engine-flutter-autoroll committed:
Roll Flutter Engine from e480e68d34a7 to 899cc167aa7c (1 revision) (#109667)

engine-flutter-autoroll committed:
Roll Flutter Engine from 899cc167aa7c to b9ba9ce9b510 (2 revisions) (#109670)

yaakovschectman committed:
Bring up new Windows start-up tests (#109618)

* Add Windows startup test

* Add new tests for Windows desktop

* Bring up complex layout tests for Win Desktop

* Bring up flutter_view startup test for Windows

* Add Platform View startup test for Windows

* Fix typo in .ci.yaml

* Make basename variable final

* Link to bug in .ci.yaml

engine-flutter-autoroll committed:
Roll Flutter Engine from b9ba9ce9b510 to 295776ac1ff3 (1 revision) (#109681)

engine-flutter-autoroll committed:
Roll Flutter Engine from 295776ac1ff3 to 5c141850e427 (1 revision) (#109686)

engine-flutter-autoroll committed:
Roll Plugins from d18079b41616 to 0d3d520952c1 (11 revisions) (#109688)

camsim99 committed:
Re-land Add Spell Check to EditableText  (#109643)

craiglabenz committed:
[docs][FWW] DropdownButton, ScaffoldMessenger, and StatefulBuilder links (#100316)

* added dropdown and scaffoldmessenger FWW links

* added StatefulBuilder wotw link

* added Autocomplete FWW link

* removed accidental auto-formatting

engine-flutter-autoroll committed:
Roll Flutter Engine from 5c141850e427 to fa316fccb372 (1 revision) (#109691)

engine-flutter-autoroll committed:
Roll Flutter Engine from fa316fccb372 to df186d820011 (1 revision) (#109699)

fluttergithubbot committed:
Marks Windows_android channels_integration_test_win to be flaky (#109695)

engine-flutter-autoroll committed:
Roll Flutter Engine from df186d820011 to ba62c6510edc (2 revisions) (#109704)

engine-flutter-autoroll committed:
Roll Flutter Engine from ba62c6510edc to 3df3a6e8eba5 (2 revisions) (#109710)

bernaferrari committed:
Fix AppBar centerTitle position with actions. (#106256)

bernaferrari committed:
Improve ShapeDecoration performance. (#108648)

pdblasi-google committed:
109638: Windows framework_tests_misc is 2.06% flaky (#109640)

engine-flutter-autoroll committed:
Roll Flutter Engine from 3df3a6e8eba5 to 6f8c99debd14 (2 revisions) (#109713)

aam committed:
Introduce stubbed `exclusive` parameter to `File.create`-overridden method (#109646)

* Introduce stubbed `exclusive` parameter to `File.create`-overridden method.

This is to soften landing of a breaking change https://github.com/dart-lang/sdk/issues/49647

gspencergoog committed:
Some misc changes needed for MenuBar implementation. (#109555)

engine-flutter-autoroll committed:
Roll Flutter Engine from 6f8c99debd14 to a91226d88602 (3 revisions) (#109716)

camsim99 committed:
Remove scheduler (#109714)

engine-flutter-autoroll committed:
Roll Flutter Engine from a91226d88602 to 4b8d1fad35cd (2 revisions) (#109719)

keyonghan committed:
Update test owners (#109717)

engine-flutter-autoroll committed:
Roll Flutter Engine from 4b8d1fad35cd to e8d5233a81b6 (1 revision) (#109725)

moffatman committed:
Don't disable pointer interaction during trackpad scroll (#106890)

engine-flutter-autoroll committed:
Roll Flutter Engine from e8d5233a81b6 to fbbd8c6bab64 (1 revision) (#109727)

engine-flutter-autoroll committed:
Roll Flutter Engine from fbbd8c6bab64 to 32d8c521da4b (3 revisions) (#109744)

engine-flutter-autoroll committed:
Roll Flutter Engine from 32d8c521da4b to 654f8029aad4 (1 revision) (#109748)

hellohuanlin committed:
[platform_view]terminate the app in tear down for platform view xcuitest (#109720)

engine-flutter-autoroll committed:
Roll Flutter Engine from 654f8029aad4 to a77ead45a80a (1 revision) (#109752)

engine-flutter-autoroll committed:
Roll Flutter Engine from a77ead45a80a to 3d85fe7ee454 (1 revision) (#109760)

engine-flutter-autoroll committed:
Roll Flutter Engine from 3d85fe7ee454 to 9807e5ce95c7 (1 revision) (#109763)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9807e5ce95c7 to 1311fea574d6 (2 revisions) (#109765)

TahaTesser committed:
Add an example for how to hide default scrollbar on desktop platform. (#108542)

yaakovschectman committed:
Add windows implementation to platform_view example (#109715)

* Add Windows platform view example

* Add new tests for Windows desktop

* Add Windows platform view

* Use const constructors (linux analyze)\

* Fix spelling/format

* Clean up code

* Remove default case

engine-flutter-autoroll committed:
Roll Plugins from 0d3d520952c1 to 4395e3195be7 (5 revisions) (#109777)

jensjoha committed:
[flutter_tools] Make flutter test -v print timing of different phases (#108864)

QuncCccccc committed:
Reland: Set IconButton.visualDensity default to VisualDensity.standard (#109432)

* Revert "Revert "Set `IconButton.visualDensity` default to `VisualDensity.standard` (#109349)" (#109421)"

Co-authored-by: Qun Cheng <quncheng@google.com>

engine-flutter-autoroll committed:
Roll Flutter Engine from 1311fea574d6 to e85e13d3d2ff (1 revision) (#109781)

engine-flutter-autoroll committed:
Roll Flutter Engine from e85e13d3d2ff to b8603c05f086 (4 revisions) (#109786)

engine-flutter-autoroll committed:
Roll Flutter Engine from b8603c05f086 to 82465fc712bc (1 revision) (#109790)

engine-flutter-autoroll committed:
Roll Flutter Engine from 82465fc712bc to c090055485cf (1 revision) (#109794)

engine-flutter-autoroll committed:
Roll Flutter Engine from c090055485cf to 50cfc83e2c9b (7 revisions) (#109814)

xu-baolin committed:
[Dismissible]Fix a state lose issue (#108971)

engine-flutter-autoroll committed:
Roll Flutter Engine from 50cfc83e2c9b to 5c30ead197b9 (6 revisions) (#109826)

tgucio committed:
Replace empty Container with const SizedBox (#108672)

engine-flutter-autoroll committed:
Roll Flutter Engine from 5c30ead197b9 to 29fe33348b9e (12 revisions) (#109860)

engine-flutter-autoroll committed:
Roll Flutter Engine from 29fe33348b9e to 7db434b2087d (1 revision) (#109862)

eyebrowsoffire committed:
Generate syntax for plugin registration that works both with and without null safety. (#109480)

Piinks committed:
Update Ballistic animation & ClampingScrollSimulation (#107735)

GaryQian committed:
Update ci.yaml tests to Android API 33 (#109820)

engine-flutter-autoroll committed:
Roll Plugins from 4395e3195be7 to cd586d7d6a40 (5 revisions) (#109866)

Commit:
Temporarily remove a bogus warning until fixed by dwds update (#109793)

* Temporarily remove a bogus warning until fixed by dwds update

* Addressed CR feedback

Commit:
Remove passing of --debugger-module-names flag to frontend server (#109791)

The flag is true by default so the behavior does not change.

Next steps:

Use the flag for updated debugger module names:
  - Frontend server: make the current behavior controlled
    by the flag non-conditional
  - Frontend server: add more debugging names changes under
    the same flag, false by default
  - Dwds: make changes required for the new module names.
  - Flutter tools: when matching dwds changes roll to flutter,
    pass the flag to the frontend server again.

- Cleanup:
  - Frontend server: make new behavior default
  - Flutter tools: remove uses of the flag.
  - Frontend server: remove the flag.

Towards: https://github.com/dart-lang/webdev/issues/1692
Helps: https://github.com/flutter/flutter/issues/106727

engine-flutter-autoroll committed:
Roll Flutter Engine from 7db434b2087d to 544c34102586 (1 revision) (#109870)

gspencergoog committed:
Re-Land "Refactor StrokeAlign to allow double values." (#109591) (#109805)

engine-flutter-autoroll committed:
Roll Flutter Engine from 544c34102586 to 92bc431e097d (3 revisions) (#109873)

thkim1011 committed:
[gen_l10n] Add option to format generated localizations files (#109171)

* init

* fix

* fix 2

* fix 3

* tests

* fix tests

* clarify help text

* fix all tests

* fix formatting?

* add second test

* unused import

* remove print

* trailing spaces

* artifacts is never null

* fix

Jasguerrero committed:
add GeneralInfo project validator to analyze --suggestions (#109874)

engine-flutter-autoroll committed:
Roll Flutter Engine from 92bc431e097d to c1ce554ac5bd (1 revision) (#109876)

keyonghan committed:
mark flaky (#109890)

dependabot[bot] committed:
Bump github/codeql-action from 2.1.18 to 2.1.19 (#109888)

hellohuanlin committed:
[platform_view]fix typo in xcuitest (#109884)

goderbauer committed:
Add some logs to catch flake (#109887)

engine-flutter-autoroll committed:
Roll Flutter Engine from c1ce554ac5bd to a418ffad89a9 (4 revisions) (#109882)

engine-flutter-autoroll committed:
Roll Flutter Engine from a418ffad89a9 to 94d07855d7a4 (3 revisions) (#109893)

engine-flutter-autoroll committed:
Roll Flutter Engine from 94d07855d7a4 to a7868e9567fc (3 revisions) (#109895)

engine-flutter-autoroll committed:
Roll Flutter Engine from a7868e9567fc to a0dcde4fb611 (1 revision) (#109897)

Commit:
[framework] remove restriction on texture layer rasterization (#109900)

engine-flutter-autoroll committed:
Roll Flutter Engine from a0dcde4fb611 to 7fb831136168 (2 revisions) (#109901)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7fb831136168 to 98c567732402 (1 revision) (#109903)

engine-flutter-autoroll committed:
Roll Flutter Engine from 98c567732402 to 835ba6b38df1 (1 revision) (#109909)

engine-flutter-autoroll committed:
Roll Flutter Engine from 835ba6b38df1 to 1e63130d7b4e (1 revision) (#109913)

engine-flutter-autoroll committed:
Roll Flutter Engine from 1e63130d7b4e to 06900c486cee (1 revision) (#109915)

engine-flutter-autoroll committed:
Roll Flutter Engine from 06900c486cee to 0005149dca9b (1 revision) (#109917)

engine-flutter-autoroll committed:
Roll Flutter Engine from 0005149dca9b to d89aae21e9fe (2 revisions) (#109922)

engine-flutter-autoroll committed:
Roll Flutter Engine from d89aae21e9fe to 00a48acb208d (1 revision) (#109928)

engine-flutter-autoroll committed:
Roll Flutter Engine from 00a48acb208d to 94c8b3ef5f38 (2 revisions) (#109931)

engine-flutter-autoroll committed:
Roll Flutter Engine from 94c8b3ef5f38 to e87ddd9819f4 (1 revision) (#109934)

engine-flutter-autoroll committed:
Roll Flutter Engine from e87ddd9819f4 to 9d831cd8ed59 (1 revision) (#109938)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9d831cd8ed59 to 3b0c87fa9f28 (2 revisions) (#109941)

engine-flutter-autoroll committed:
Roll Flutter Engine from 3b0c87fa9f28 to d030ae1a5a67 (1 revision) (#109944)

gspencergoog committed:
Revert "Re-Land "Refactor StrokeAlign to allow double values." (#109805)" (#109955)

This reverts commit b3aba4d9c868aa5a5f8c3a518960f454eca2ba40 because it breaks Google tests.

zanderso committed:
[flutter_tools] Enable custom devices on all channels (#109953)

engine-flutter-autoroll committed:
Roll Flutter Engine from d030ae1a5a67 to bd89852694fe (1 revision) (#109961)

engine-flutter-autoroll committed:
Roll Flutter Engine from bd89852694fe to ebee9c3fb865 (3 revisions) (#109967)

engine-flutter-autoroll committed:
Roll Flutter Engine from ebee9c3fb865 to d311584c791d (1 revision) (#109972)

engine-flutter-autoroll committed:
Roll Flutter Engine from d311584c791d to 7295b1ee8765 (1 revision) (#109982)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7295b1ee8765 to 482f967fa6b2 (1 revision) (#109987)

engine-flutter-autoroll committed:
Roll Flutter Engine from 482f967fa6b2 to b9d670aaddcf (1 revision) (#109990)

engine-flutter-autoroll committed:
Roll Flutter Engine from b9d670aaddcf to 5f732d4e4e7a (1 revision) (#109992)

engine-flutter-autoroll committed:
Roll Flutter Engine from 5f732d4e4e7a to 00bb6056cb80 (1 revision) (#109995)

engine-flutter-autoroll committed:
Roll Flutter Engine from 00bb6056cb80 to e70ba6a3a32d (1 revision) (#110003)

bryanoltman committed:
Add clipBehavior to Card+InkWell example (#109872)

* Add clipBehavior to Card+InkWell example

* Add test for example

* Update comment

engine-flutter-autoroll committed:
Roll Plugins from cd586d7d6a40 to 89fe2a2c0b09 (4 revisions) (#110008)

engine-flutter-autoroll committed:
Roll Flutter Engine from e70ba6a3a32d to 0d182ddb0d1d (1 revision) (#110009)

hannah-hyj committed:
InputDecorator iconColor/prefixIconColor/suffixIconColor (#109988)

engine-flutter-autoroll committed:
Roll Flutter Engine from 0d182ddb0d1d to 60707621bf9b (1 revision) (#110011)

gspencergoog committed:
Remove the FocusScopeNode in the navigator (#109702)

engine-flutter-autoroll committed:
Roll Flutter Engine from 60707621bf9b to ca7acb34c5bc (2 revisions) (#110015)

engine-flutter-autoroll committed:
Roll Flutter Engine from ca7acb34c5bc to 38d2213d99ac (1 revision) (#110026)

engine-flutter-autoroll committed:
Roll Flutter Engine from 38d2213d99ac to 9513c34aabcd (2 revisions) (#110033)

Piinks committed:
Deprecate 2018 text theme parameters (#109817)

engine-flutter-autoroll committed:
0fd04455f deprecate pushPhysicalLayer (flutter/engine#35566) (#110036)

dependabot[bot] committed:
Bump github/codeql-action from 2.1.19 to 2.1.20 (#110044)

engine-flutter-autoroll committed:
Roll Flutter Engine from 0fd04455fd77 to 7252c6406b2c (3 revisions) (#110042)

zanderso committed:
Remove most benchmarks of SkSL warmup (#110040)

QuncCccccc committed:
Fixed leading button size on app bar (#110043)

ricardoamador committed:
Moving ios benchmark tests to prod. (#110049)

* Moving io benchmark tests to prod.

* New tasks must be marked bringup: true

hellohuanlin committed:
[platform_view]retry launch if it fails to launch for xcuitest (#110030)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7252c6406b2c to 258c580a83c6 (3 revisions) (#110051)

engine-flutter-autoroll committed:
Roll Flutter Engine from 258c580a83c6 to ca48808d06f3 (1 revision) (#110054)

engine-flutter-autoroll committed:
Roll Flutter Engine from ca48808d06f3 to 8c29591b89a3 (10 revisions) (#110078)

engine-flutter-autoroll committed:
Roll Flutter Engine from 8c29591b89a3 to c9d0a012817d (1 revision) (#110087)

engine-flutter-autoroll committed:
Roll Plugins from 89fe2a2c0b09 to 3a5e6f3acdd3 (4 revisions) (#110092)

keyonghan committed:
Copy artifacts to `applicationBinaryPath` when specified for build+test separation (#109879)

yaakovschectman committed:
Document tristate value (#110106)

Documentation for Checkbox.value previously did not cover the tristate case.

QuncCccccc committed:
Revert "Fixed leading button size on app bar (#110043)" (#110103)

This reverts commit 7e33cd5ef30b9960d5c8d6fc75b42dabb7ec1f2f.

keyonghan committed:
Update open_jdk version `version:11` (#110110)

QuncCccccc committed:
Revert "Reland: Set IconButton.visualDensity default to VisualDensity.standard (#109432)" (#110119)

This reverts commit 79013b2be8d5e726bb4447bce665f63b5eb9308a.

engine-flutter-autoroll committed:
Roll Flutter Engine from c9d0a012817d to 8cc02dc73e95 (11 revisions) (#110121)

* 94f97c337 Roll Skia from b91dbe274717 to 94602de608e0 (4 revisions) (flutter/engine#35634)

* 6ec08c50e Roll Skia from 94602de608e0 to 8c32b0148770 (7 revisions) (flutter/engine#35636)

* 9195b7a2f Roll Dart SDK from 03506589f2d0 to c064b9bacc4d (1 revision) (flutter/engine#35637)

* f7ec70782 Roll Skia from 8c32b0148770 to a9a05d0bf0c5 (3 revisions) (flutter/engine#35639)

* 15cf53acc Image shader dispose (flutter/engine#35640)

* ee3cd0d8a Roll Skia from a9a05d0bf0c5 to 147232a3cebc (2 revisions) (flutter/engine#35641)

* 5f53ee3b2 [Impeller] Use linear sampling for reads in the texture atlas (flutter/engine#35618)

* 4ff7f7b9a Roll Fuchsia Mac SDK from JocShsJvXYeSVjgvW... to Es8SXkqDOYZ5dgYTq... (flutter/engine#35642)

* 565cc861e Roll Skia from 147232a3cebc to cc63d5dd13b6 (1 revision) (flutter/engine#35643)

* b4af69f02 Roll Dart SDK from c064b9bacc4d to 3042d6196824 (1 revision) (flutter/engine#35644)

* 8cc02dc73 Roll Skia from cc63d5dd13b6 to a4744aa054cf (2 revisions) (flutter/engine#35645)

engine-flutter-autoroll committed:
Roll Flutter Engine from 8cc02dc73e95 to 9dee4b9e4b8f (1 revision) (#110122)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9dee4b9e4b8f to 2ecccf195ec7 (2 revisions) (#110128)

* 68beed75a Refactor - Create `Renderer` interface for CanvasKit/HTML switching (flutter/engine#35502)

* 2ecccf195 [Impeller] Enforce a minimum stroke width (flutter/engine#35576)

engine-flutter-autoroll committed:
Roll Flutter Engine from 2ecccf195ec7 to de4ef8c6f379 (3 revisions) (#110132)

* d1ac8d84e Move extension to TypefaceFontProvider instead of SkFontMgr (flutter/engine#35381)

* 55b46a70a Roll Skia from fad2781bd328 to 6cec033662c7 (3 revisions) (flutter/engine#35649)

* de4ef8c6f Roll Dart SDK from 3042d6196824 to 9be467274ae4 (1 revision) (flutter/engine#35650)

wer-mathurin commented:
@Hixie
 I will look into that.

engine-flutter-autoroll committed:
7d0afb03e Roll Clang Windows from 60d276923902 to 41d4067d0b7c (flutter/engine#35655) (#110135)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7d0afb03e363 to 59a61b2300ea (2 revisions) (#110137)

Commit:
Merge branch 'feat/paginated_data_table_linear_progress' of https://github.com/wer-mathurin/flutter into feat/paginated_data_table_linear_progress

Commit:
Merge branch 'feat/paginated_data_table_linear_progress' of https://github.com/wer-mathurin/flutter into feat/paginated_data_table_linear_progress

Commit:
Merge branch 'feat/paginated_data_table_linear_progress' of https://github.com/wer-mathurin/flutter into feat/paginated_data_table_linear_progress

Commit:
Revert "FIX - Specify Type Annotation"

This reverts commit 3c6ecb2572d470a96c3594d0479b0db0ce4675f2.

Commit:
Revert "FIX - Specify Type Annotation"

This reverts commit 3c6ecb2572d470a96c3594d0479b0db0ce4675f2.

Commit:
Merge branch 'feat/paginated_data_table_linear_progress' of https://github.com/wer-mathurin/flutter into feat/paginated_data_table_linear_progress
